src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1841,16 +1816,25 @@ void SolverInterfaceImpl::mapWrittenData()
   }
 }
 
-void SolverInterfaceImpl::mapReadData()
+void SolverInterfaceImpl::mapReadDataAndPerformReadAction(std::vector<double> receiveTimes)
```

Comment:
IMO, this function (name) is overkill. But let's re-evaluate once we have a symmetric treatment of read and write.